client/components/auto-direction/test/auto-direction.jsx

```diff
@@ -1,31 +1,29 @@
 /**
  * @jest-environment jsdom
  */
-
-import { expect } from 'chai';
-import { shallow } from 'enzyme';
+import { render, screen } from '@testing-library/react';
 import AutoDirection from '..';
 
 describe( 'AutoDirection', () => {
 	describe( 'component rendering', () => {
 		test( 'adds a direction to RTL text', () => {
-			const wrapper = shallow(
+			render(
 				<AutoDirection>
-					<div>השנה היא 2017.</div>
+					<div data-testid="direction-test">השנה היא 2017.</div>
 				</AutoDirection>
 			);
 
-			expect( wrapper.getElement().props.direction ).to.equal( 'rtl' );
+			expect( screen.getByTestId( 'direction-test' ).getAttribute( 'direction' ) ).toEqual( 'rtl' );
```

Comment:
Interesting! What's the difference between using `screen` and storing a reference to the compnent?

Reply:
Good question! 

First of all, `screen.getByTestId( 'x' )` is essentially a shortcut to `container.querySelector( '[data-testid="x"]')`, but the former more idiomatic so I generally prefer it.  

Furthermore, according to the [first guiding principle](https://testing-library.com/docs/guiding-principles/) of `@testing-library`:

> If it relates to rendering components, then it should deal with DOM nodes rather than component instances, and it should not encourage dealing with component instances.

So when testing, we try to avoid component instances and accessing props; rather we access the DOM.

Finally, according to the `*ByTestId()` [docs](https://testing-library.com/docs/queries/bytestid/):

> it is recommended to use this only after the other queries don't work for your use case. Using data-testid attributes do not resemble how your software is used and should be avoided if possible. 

So my goal here was to both:

* use a `test-id` for testing demonstration and education purposes, while:
* still not having to add it to a component when that's not really necessary.

Let me know if that makes sense 😉

Reply:
Very interesting, thanks! I think I didn't understand quite what `container` was:

> The containing DOM node of your rendered React Element (rendered using ReactDOM.render). It's a div. This is a regular DOM node, so you can call container.querySelector etc. to inspect the children.
> ...
>  If you find yourself using container to query for rendered elements then you should reconsider! The other queries are designed to be more resilient to changes that will be made to the component you're testing. Avoid using container to query for elements!

So it's not a reference to a component, but a DOM node. And then returned along with `container` are several testing library queries bound to that container DOM node. Instead of using those, you can just use `screen`, which automatically binds all the queries to the document. 👍